confluence-xml/src/main/java/org/xwiki/contrib/confluence/filter/internal/input/ConfluenceInputFilterStream.java

```diff
@@ -231,25 +271,14 @@ private void readInternal(Object filter, ConfluenceFilter proxyFilter) throws Fi
             // < WikiSpace
             proxyFilter.endWikiSpace(spaceKey, spaceParameters);
         }
-
-        // Generate Blog events
-        if (this.properties.isBlogsEnabled()) {
-            generateBlogEvents(blogPages, filter, proxyFilter);
-        }
-
-        this.progress.popLevelProgress(this);
-
-        // Cleanup
-
-        observationManager.notify(new ConfluenceFilteredEvent(), this, this.confluencePackage);
-
-        closeConfluencePackage();
     }
 
     private void sendPages(Object filter, ConfluenceFilter proxyFilter, Map.Entry<Long, List<Long>> entry, String spaceKey)
     {
-        for (long pageId : entry.getValue()) {
-            sendPage(pageId, spaceKey, filter, proxyFilter, false);
+        if (this.properties.isContentsEnabled() || this.properties.isRightsEnabled()) {
```

Comment:
I find a bit strange to import page rights if pages are not supposed to be imported.

Reply:
I still don't see the relationship, if a user wants the pages again the pages should not be disabled. What is sure is that just disabling the page should be enough to not have the pages.

Keep in mind that what you send is not a diff, it's the complete document import, so if you produce events with a page having an empty content you will end up with a page having empty content overwriting whatever was the previous content.